akka-actor/src/main/scala/akka/io/TcpConnection.scala

```diff
@@ -335,6 +335,10 @@ private[io] abstract class TcpConnection(val tcp: TcpExt, val channel: SocketCha
         if (TraceLogging) log.debug("setSoLinger(true, 0) failed with [{}]", e)
     }
     channel.close()
+    // On linux, closing sends the RST directly. On windows, however, the connection is
+    // closed only when the channel is deregistered from the selector, which happens before
+    // and after 'select' polling.
```

Comment:
I looked at/stepped through the JVM sources: `java.nio.channels.spi.AbstractSelector` contains a list of `cancelledKeys`, and `sun.nio.ch.SelectorImpl` will `kill` those from `processDeregisterQueue` which is called before and after `select()`.

This is the same between linux and windows, but on linux we don't need to rely on the `kill` to close the connection, as it has already been closed as a side effect of `preClose` called from `sun.nio.ch.SocketChannelImpl#implCloseSelectableChannel`.